src/parser/core/metric.ts

```diff
@@ -0,0 +1,28 @@
+import { AnyEvent } from './Events';
+import Ability from './modules/Ability';
+
+export interface Info {
+  playerId: number;
+  abilities: Ability[];
+  fightStart: number;
+  fightEnd: number;
+}
+export type Metric<Value = any> = (events: AnyEvent[], info: Info, ...args: any[]) => Value;
+
+const metric = <T extends any[], U>(fn: (events: AnyEvent[], ...args: T) => U) => {
```

Comment:
Seems like just what we need - for the sake of future devs could add docs explaining what this is and when it should be used?